### DISCUSSION TITLE
How to get ioregistry info?

### DISCUSSION THREAD

**Discussion by Hoangdus** [doronz88/pymobiledevice3#427] [Q&A]
I was trying to get more diagnostics data using ```python3 -m pymobiledevice3 diagnostics ioregistry``` but for some reason it returned NULL. Am I missing something or what did I do wrong? Thanks in advance.

**Comment by doronz88** [comment]
If no results then the ioregistry query just returned an empty set.
You should supply additional parameters regarding the `plane` and/or `name` of what you're looking for.

**Reply by Hoangdus** [reply]
I just tried ```python3 -m pymobiledevice3 diagnostics ioregistry --plane IOService --udid 'my udid'``` and it didn't work it returned `{}` and also when I pass in `PYMOBILEDEVICE3_UDID` instead of my device udid it shows a usbmux error not sure if that's normal. My device is running iOS 15.1

**Reply by doronz88** [reply]
You just confused the query a bit
```shell
python3 -m pymobiledevice3 diagnostics ioregistry --ioclass IOService
```

**Reply by Hoangdus** [reply]
Oh, thanks it works now but IOPower is still return ```{}``` I know it's a plane so I use ```--plane``` instead of ```--ioclass```

**Reply by doronz88** [reply]
Sorry but I couldn't figure out why the query returns an empty resultset. That's just how the device chooses to answer
